src/librawspeed/metadata/ColorFilterArray.h

```diff
@@ -27,16 +27,15 @@
 namespace RawSpeed {
 
 enum CFAColor {
-  CFA_COLOR_MIN = 0,
+  // see also DngDecoder
   CFA_RED = 0,
   CFA_GREEN = 1,
   CFA_BLUE = 2,
-  CFA_GREEN2 = 3,
+  CFA_INVALID = 3, // keep the unit tests happy
```

Comment:
> So the 'correct' approach would be to remove the constructor taking a dcraw_filter and instead provide a little helper function that does perform the bit-swiveling directly on the uint32. 

That is more or less how it was done initially
The shifted CFA's were manually pre-computed, and then just returned based on the shift value. It worked ok when dt was ignorant about non 3-color bayer images. But with support for CYGM, it was replaced with this "constructor taking a dcraw_filter" stuff.

I don't really care about "constructor taking a dcraw_filter". But i do care for such CFA shift to work. So i'd be ok with `instead provide a little helper function that does perform the bit-swiveling directly on the uint32.`